src/vdlm2.rs

```diff
@@ -181,39 +177,39 @@ pub struct AvlcData {
     #[serde(skip_serializing_if = "Option::is_none")]
     pub poll: Option<bool>,
     #[serde(skip_serializing_if = "Option::is_none")]
-    pub acars: Option<AvlcAcars>
+    pub acars: Option<AvlcAcars>,
 }
 
 #[derive(Debug, Clone, Serialize, Deserialize, PartialEq, PartialOrd)]
 pub struct DstBlock {
     pub addr: String,
     #[serde(rename = "type")]
-    pub vehicle_type: String
+    pub vehicle_type: String,
 }
 
 #[derive(Debug, Clone, Serialize, Deserialize, PartialEq, PartialOrd)]
 pub struct SrcBlock {
     pub addr: String,
     pub status: String,
     #[serde(rename = "type")]
-    pub source_type: String
+    pub source_type: String,
 }
 
 #[derive(Debug, Clone, Serialize, Deserialize, PartialEq, PartialOrd)]
 pub struct XidBlock {
     pub err: bool,
-    pub pub_params: Vec<XidParam>,
+    pub pub_params: Option<Vec<XidParam>>,
```

Comment:
So for these you'll want to add in the `#[serde(skip_serializing_if = "Option::is_none")]` preceding each of the entries you've now made `Optional<>` here. Forces them to be ignored when converting from Object to String so that you don't potentially end up with a `"pub_params": None` being inserted into the payload.